testing.rb

```diff
@@ -0,0 +1,311 @@
+          total = yield(total, arr[sum])
+          sum += 1
+          num += 1
+        end
+        total
+      end
+    elsif block_given? && !args.nil? && arg.nil? # blocks and args
+      sum = 1
+      num = 0
+      totals = arr[num]
+      while sum < arr.length
+        totals = yield(totals, arr[sum])
+        sum += 1
+        num += 1
+      end
+      yield(totals, args)
+    elsif !block_given? && !args.nil? && !arg.nil? # no blocks and args
+      i = 1
+      x = 0
+      acum = arr[x]
+      while i < arr.length
+        acum = acum.send(arg, arr[i])
+        i += 1
+      end
+      acum.send(arg, args)
+    elsif !block_given? && args.nil? && args.nil? # no blocck and no args
+      yield
+    end
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |item| return false unless yield(item) }
+      true
+
+    elsif !block_given? && !arg.nil?
+      if arg.class == Regexp
+        my_each { |item| return false unless item.match?(arg) }
+        true
+
+      elsif arg == Numeric
+        my_each { |item| return false if item.class.superclass != arg }
+        true
+
+      elsif arg == Integer
+        my_each { |item| return false if item.class != arg }
+        true
+      elsif !arg.nil?
+        my_each { |item| return false if item != arg }
+        true
+      end
+    elsif !block_given? && arg.nil?
+      my_each { |item| return false if item.nil? || item == false }
+      true
+    elsif !block_given? && empty?
+      true
+    end
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      false
+
+    elsif !block_given? && !arg.nil?
+      if arg.class == Regexp
+        my_each  do |item|
+          char = item.split('')
+          char.my_each do |n|
+            return true if n.match?(arg)
+          end
+        end
+        false
+      elsif arg.class == String
+        my_each  do |item|
+          char = item.split(' ')
+          char.my_each do |n|
+            return true if n.match?(arg)
+          end
+        end
+        false
+      elsif arg == Integer
+        my_each { |item| return true if item.class == arg }
+        false
+      elsif arg == Numeric
+        my_each { |item| return true if item.class != arg }
+        false
+      end
+    elsif !block_given? && arg.nil?
+      my_each { |item| return true if item == true }
+      false
+    elsif !block_given? && empty?
+      true
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+
+    array2 = []
+    arr.my_each do |item|
+      array2.push(item) if yield(item)
+    end
+    array2
+  end
+
+  def multiply_els(args = nil, arg = nil)
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    if !block_given? && !args.nil? && arg.nil? # no blocks and args
+      i = 1
+      x = 0
+      acum = arr[x]
```

Comment:
Your #multiply_els accepts an **array as an argument** and multiplies all the elements of the array together using **#my_inject**  So you should have something like `def multiply_els(arr)`

You don't need all those codes inside your #multiply_els method, call the #my_inject method inside.